### PR TITLE
Add threshold option to compression.

### PR DESCRIPTION
**Pull request by Nibbler999**
I think we need an option like this to limit compression to larger payloads. The default value of 1024 bytes is taken from socket.io.

### CODE REVIEW

File: lib/Sender.js [hunk 1]
```diff
@@ -142,6 +142,11 @@ class Sender {
    * @api private
    */
   sendCompressed (opcode, data, finalFragment, mask, compress, cb) {
+    if (data && data.length < this.extensions[PerMessageDeflate.extensionName].threshold) {
```
Comment (Nibbler999) [inline lib/Sender.js]: We have to enqueue to preserve message ordering. I'll think about `length`.

File: lib/Sender.js [hunk 2]
```diff
@@ -78,7 +78,7 @@ class Sender {
    */
   doPing (data, options) {
     var mask = options && options.mask;
-    this.frameAndSend(0x9, data || '', true, mask);
+    this.frameAndSend(0x9, data ? Buffer.from(data.toString()) : null, true, mask);
```
Comment (lpinca) [inline lib/Sender.js]: Is `toString()` to make sure that whatever is passed to `ping` is converted to a string?

File: test/WebSocket.test.js [hunk 3]
```diff
@@ -2226,7 +2226,7 @@ describe('WebSocket', function() {
     describe('#terminate', function() {
       it('will raise error callback, if any, if called during send data', function(done) {
         var wss = new WebSocketServer({port: ++port, perMessageDeflate: true}, function() {
-          var ws = new WebSocket('ws://localhost:' + port, {perMessageDeflate: true});
+          var ws = new WebSocket('ws://localhost:' + port, {perMessageDeflate: { threshold: 0 }});
```
Comment (lpinca) [inline test/WebSocket.test.js]: Why this test fails without this change?

File: lib/Sender.js (hunk 2 above)
Comment (Nibbler999) [inline lib/Sender.js]: Yes, we need to be able to accept a number here.

File: test/WebSocket.test.js (hunk 3 above)
Comment (Nibbler999) [inline test/WebSocket.test.js]: The test relies on compression but the payload is below the default threshold size.

File: lib/Sender.js (hunk 2 above)
Comment (lpinca) [inline lib/Sender.js]: Ok, It might be better to use `String(data)` which won't fail with `null` or `undefined`.

**Pull request merged**